test/evp_extra_test2.c

```diff
@@ -1073,6 +1073,36 @@ static int test_evp_md_ctx_copy(void)
     return ret;
 }
 
+#if !defined OPENSSL_NO_DES && !defined OPENSSL_NO_MD5
+static int test_evp_pbe_alg_add(void)
+{
+    int ret = 0;
+    int cipher_nid = 0, md_nid = 0;
+    EVP_PBE_KEYGEN_EX *keygen_ex = NULL;
+    EVP_PBE_KEYGEN *keygen = NULL;
+
+
+    if (!TEST_true(EVP_PBE_alg_add(NID_pbeWithMD5AndDES_CBC, EVP_des_cbc(), EVP_md5(),
+                    PKCS5_PBE_keyivgen)))
+        goto err;
+
+    if (!TEST_true(EVP_PBE_find_ex(EVP_PBE_TYPE_OUTER, NID_pbeWithMD5AndDES_CBC,
+                         &cipher_nid, &md_nid, &keygen, &keygen_ex)))
+
+        goto err;
+
+    if (!TEST_true(keygen != NULL))
```

Comment:
`if (!TEST_ptr(keygen))`

Reply:
@paulidale @hlandau @levitte ISO C does not allow to convert functional pointer to `(void*)`, so `TEST_ptr(.)` and `TEST_ptr_null(.)` macros do not work. My idea was to introduce macros `TEST_func_ptr(.)`, `TEST_func_ptr_null(.)`.